Node v4+ NOT compatibility

```bash
$ node -v
v6.9.1
$ npm test

> linez@4.1.1 test /mnt/e/work/linez
> gulp test

[15:19:18] Using gulpfile /mnt/e/work/linez/gulpfile.js
[15:19:18] Starting 'clean'...
[15:19:18] Finished 'clean' after 177 μs
[15:19:18] Starting 'clean'...
[15:19:18] Starting 'tslint'...
[15:19:18] Finished 'tslint' after 578 ms
[15:19:18] Finished 'clean' after 593 ms
[15:19:18] Starting 'scripts'...
[15:19:20] Finished 'scripts' after 1.42 s
[15:19:20] Starting 'build'...
[15:19:20] Finished 'build' after 2 μs
[15:19:20] Starting 'test'...


  StringFinder
    constructor
      ✓ requires a string[] or Regex in the constructor
      ✓ requires regular expression to have the global flag
    findAll method
      ✓ finds 2 needles in a haystack

  linez
    ✓ parses empty text
    ✓ parses a single line with no line ending
    ✓ parses a single line with a line ending
    ✓ sets proper line offsets
    ✓ sets proper line numbers
    ✓ sets proper line text
    ✓ sets proper line endings
    ✓ supports resetting the configuration
    configure method
      ✓ supports a custom newlines string array
      ✓ errors when no configuration options are sent
    Document
      ✓ constructs a document with lines
      ✓ converts doc into a buffer with toBuffer() (66ms)
      ✓ converts a signed doc into a buffer with toBuffer()
      ✓ converts an unsigned doc into a string with toString()
      ✓ throws when trying to set an unsupported charset
      byte order marks
        ✓ detects and decodes a utf-8-bom document
        ✓ detects and decodes utf-16le bom document
        1) detects and decodes utf-16be bom document
        ✓ throws an unsupported charset error for utf-32le bom
        ✓ throws an unsupported charset error for utf-32be bom
        ✓ decodes unsigned docs as utf8 by default


  23 passing (122ms)
  1 failing

  1) linez Document byte order marks detects and decodes utf-16be bom document:

      AssertionError: expected '뢸숃\u0000' to equal 'foo'
      + expected - actual

      -뢸숃
      +foo

      at Context.<anonymous> (js/linez.spec.js:9:9230)



[15:19:20] 'test' errored after 566 ms
[15:19:20] Error in plugin 'gulp-mocha'
Message:
    1 test failed.
npm ERR! Test failed.  See above for more details.
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,8 +1,10 @@
 language: node_js
 
 node_js:
-  - '0.12'
-  - '0.10'
+  - stable
+  - 6
+  - 4
+  - 0
 
 sudo: false
 

--- a/js/StringFinder.js
+++ b/js/StringFinder.js
@@ -1,3 +1,4 @@
+"use strict";
 var StringFinder = (function () {
     function StringFinder(needles) {
         if (needles instanceof RegExp) {
@@ -34,5 +35,5 @@ var StringFinder = (function () {
         return matches;
     };
     return StringFinder;
-})();
+}());
 module.exports = StringFinder;

--- a/js/StringFinder.spec.js
+++ b/js/StringFinder.spec.js
@@ -1,3 +1,4 @@
+"use strict";
 var sinonChai = require('./test-common');
 var expect = sinonChai.expect;
 var StringFinder = require('./StringFinder');
@@ -11,12 +12,8 @@ describe('StringFinder', function () {
                     new StringFinder(['foo', 'bar'])
                 ];
             }).not.to.throw;
-            expect(function () {
-                new StringFinder(void (0));
-            }).to.throw('Unexpected type in StringFinder constructor argument: undefined');
-            expect(function () {
-                new StringFinder(2);
-            }).to.throw('Unexpected type in StringFinder constructor argument: number');
+            expect(function () { new StringFinder(void (0)); }).to.throw('Unexpected type in StringFinder constructor argument: undefined');
+            expect(function () { new StringFinder(2); }).to.throw('Unexpected type in StringFinder constructor argument: number');
         });
         it('requires regular expression to have the global flag', function () {
             var fn = function () {

--- a/js/linez.js
+++ b/js/linez.js
@@ -1,3 +1,4 @@
+"use strict";
 var iconv = require('iconv-lite');
 var bufferEquals = require('buffer-equals');
 var StringFinder = require('./StringFinder');
@@ -20,8 +21,8 @@ function linez(file) {
     doc.charset = detectCharset(buffer);
     var bom = boms[doc.charset];
     var encoding = doc.charset.replace(/bom$/, '');
-    if (Buffer.isEncoding(encoding)) {
-        doc.lines = parseLines(buffer.slice(bom.length).toString(encoding));
+    if (iconv.encodingExists(encoding)) {
+        doc.lines = parseLines(iconv.decode(buffer.slice(bom.length), encoding));
     }
     else {
         doc.lines = parseLines(buffer.toString('utf8'));
@@ -87,7 +88,7 @@ var linez;
                     this._charset = '';
                     return;
                 }
-                if (!Buffer.isEncoding(value.replace(/-bom$/, ''))) {
+                if (!iconv.encodingExists(value.replace(/-bom$/, ''))) {
                     throw new Error('Unsupported charset: ' + value);
                 }
                 this._charset = value;
@@ -97,9 +98,15 @@ var linez;
         });
         Document.prototype.toBuffer = function () {
             var charset = this.charset.replace(/-bom$/, '');
-            var contents = new Buffer(this.toString(), charset);
+            var contents;
+            if (iconv.encodingExists(charset)) {
+                contents = iconv.encode(this.toString(), charset);
+            }
+            else {
+                contents = new Buffer(this.toString());
+            }
             if (this.bom) {
-                return Buffer.concat([this.bom, contents]);
+                contents = Buffer.concat([this.bom, contents]);
             }
             return contents;
         };
@@ -109,7 +116,7 @@ var linez;
             }).join('');
         };
         return Document;
-    })();
+    }());
     linez.Document = Document;
     function configure(options) {
         if (!options) {

--- a/js/linez.spec.js
+++ b/js/linez.spec.js
@@ -1,4 +1,6 @@
+"use strict";
 var iconv = require('iconv-lite');
+var bufferEquals = require('buffer-equals');
 var sinonChai = require('./test-common');
 var linez = require('./linez');
 var expect = sinonChai.expect;
@@ -92,15 +94,15 @@ describe('linez', function () {
         it('converts doc into a buffer with toBuffer()', function () {
             var contents = Buffer.concat([
                 new Buffer([0xef, 0xbb, 0xbf]),
-                new Buffer('foo', 'utf8')
+                new Buffer('foo')
             ]);
             var doc = linez(contents);
-            expect(doc.toBuffer().equals(contents)).to.be.true;
+            expect(bufferEquals(doc.toBuffer(), contents)).to.be.true;
         });
         it('converts a signed doc into a buffer with toBuffer()', function () {
-            var contents = new Buffer('foo', 'utf8');
+            var contents = new Buffer('foo');
             var doc = linez(contents);
-            expect(doc.toBuffer().equals(contents)).to.be.true;
+            expect(bufferEquals(doc.toBuffer(), contents)).to.be.true;
         });
         it('converts an unsigned doc into a string with toString()', function () {
             var contents = 'foo\nbar';
@@ -118,7 +120,7 @@ describe('linez', function () {
             it('detects and decodes a utf-8-bom document', function () {
                 var doc = linez(Buffer.concat([
                     new Buffer([0xef, 0xbb, 0xbf]),
-                    new Buffer('foo', 'utf8')
+                    new Buffer('foo')
                 ]));
                 expect(doc.charset).to.eq('utf-8-bom');
                 expect(doc.lines[0].text).to.eq('foo');
@@ -126,7 +128,7 @@ describe('linez', function () {
             it('detects and decodes utf-16le bom document', function () {
                 var doc = linez(Buffer.concat([
                     new Buffer([0xff, 0xfe]),
-                    new Buffer('foo', 'utf16le')
+                    iconv.encode('foo', 'utf16le')
                 ]));
                 expect(doc.charset).to.eq('utf-16le');
                 expect(doc.lines[0].text).to.eq('foo');
@@ -134,7 +136,7 @@ describe('linez', function () {
             it('detects and decodes utf-16be bom document', function () {
                 var doc = linez(Buffer.concat([
                     new Buffer([0xfe, 0xff]),
-                    new Buffer('foo', 'utf16be')
+                    iconv.encode('foo', 'utf16be')
                 ]));
                 expect(doc.charset).to.eq('utf-16be');
                 expect(doc.lines[0].text).to.eq('foo');
@@ -152,7 +154,7 @@ describe('linez', function () {
                 expect(fn).to.throw('Unsupported charset: utf-32be');
             });
             it('decodes unsigned docs as utf8 by default', function () {
-                var doc = linez(new Buffer('foo', 'utf8'));
+                var doc = linez(new Buffer('foo'));
                 expect(doc.charset).to.be.empty;
                 expect(doc.lines[0].text).to.eq('foo');
             });

--- a/js/test-common.js
+++ b/js/test-common.js
@@ -1,4 +1,5 @@
 ///<reference path='../typings/tsd.d.ts'/>
+"use strict";
 var chai = require('chai');
 var sinonChai = require('sinon-chai');
 chai.use(sinonChai);

--- a/lib/linez.spec.ts
+++ b/lib/linez.spec.ts
@@ -5,7 +5,6 @@ import sinonChai = require('./test-common');
 import linez = require('./linez');
 
 var expect = sinonChai.expect;
-iconv.extendNodeEncodings();
 
 // ReSharper disable WrongExpressionStatement
 describe('linez', () => {
@@ -110,14 +109,14 @@ describe('linez', () => {
 		it('converts doc into a buffer with toBuffer()', () => {
 			var contents = Buffer.concat([
 				new Buffer([0xef, 0xbb, 0xbf]),
-				new Buffer('foo', 'utf8')
+				new Buffer('foo')
 			]);
 			var doc = linez(contents);
 			expect(bufferEquals(doc.toBuffer(), contents)).to.be.true;
 		});
 
 		it('converts a signed doc into a buffer with toBuffer()', () => {
-			var contents = new Buffer('foo', 'utf8');
+			var contents = new Buffer('foo');
 			var doc = linez(contents);
 			expect(bufferEquals(doc.toBuffer(), contents)).to.be.true;
 		});
@@ -141,7 +140,7 @@ describe('linez', () => {
 			it('detects and decodes a utf-8-bom document', () => {
 				var doc = linez(Buffer.concat([
 					new Buffer([0xef, 0xbb, 0xbf]),
-					new Buffer('foo', 'utf8')
+					new Buffer('foo')
 				]));
 				expect(doc.charset).to.eq('utf-8-bom');
 				expect(doc.lines[0].text).to.eq('foo');
@@ -150,7 +149,7 @@ describe('linez', () => {
 			it('detects and decodes utf-16le bom document', () => {
 				var doc = linez(Buffer.concat([
 					new Buffer([0xff, 0xfe]),
-					new Buffer('foo', 'utf16le')
+					iconv.encode('foo', 'utf16le')
 				]));
 				expect(doc.charset).to.eq('utf-16le');
 				expect(doc.lines[0].text).to.eq('foo');
@@ -159,7 +158,7 @@ describe('linez', () => {
 			it('detects and decodes utf-16be bom document', () => {
 				var doc = linez(Buffer.concat([
 					new Buffer([0xfe, 0xff]),
-					new Buffer('foo', 'utf16be')
+					iconv.encode('foo', 'utf16be')
 				]));
 				expect(doc.charset).to.eq('utf-16be');
 				expect(doc.lines[0].text).to.eq('foo');
@@ -180,12 +179,11 @@ describe('linez', () => {
 			});
 
 			it('decodes unsigned docs as utf8 by default', () => {
-				var doc = linez(new Buffer('foo', 'utf8'));
+				var doc = linez(new Buffer('foo'));
 				expect(doc.charset).to.be.empty;
 				expect(doc.lines[0].text).to.eq('foo');
 			});
 
 		});
 	});
-
 });

--- a/lib/linez.ts
+++ b/lib/linez.ts
@@ -4,7 +4,6 @@ var bufferEquals = require('buffer-equals');
 import StringFinder = require('./StringFinder');
 
 var lineEndingFinder: StringFinder;
-iconv.extendNodeEncodings();
 
 var boms: { [key: string]: Buffer } = {
 	'utf-8-bom': new Buffer([0xef, 0xbb, 0xbf]),
@@ -27,8 +26,8 @@ function linez(file: string|Buffer): linez.Document {
 	doc.charset = detectCharset(buffer);
 	var bom = boms[doc.charset];
 	var encoding = doc.charset.replace(/bom$/, '');
-	if ((<any>Buffer).isEncoding(encoding)) {
-		doc.lines = parseLines(buffer.slice(bom.length).toString(encoding));
+	if (iconv.encodingExists(encoding)) {
+		doc.lines = parseLines(iconv.decode(buffer.slice(bom.length), encoding));
 	} else {
 		doc.lines = parseLines(buffer.toString('utf8'));
 	}
@@ -92,7 +91,7 @@ module linez {
 				this._charset = '';
 				return;
 			}
-			if (!(<any>Buffer).isEncoding(value.replace(/-bom$/, ''))) {
+			if (!iconv.encodingExists(value.replace(/-bom$/, ''))) {
 				throw new Error('Unsupported charset: ' + value);
 			}
 			this._charset = value;
@@ -108,9 +107,14 @@ module linez {
 
 		toBuffer() {
 			var charset = this.charset.replace(/-bom$/, '');
-			var contents = new Buffer(this.toString(), charset);
+			var contents: Buffer;
+			if (iconv.encodingExists(charset)) {
+				contents = iconv.encode(this.toString(), charset);
+			} else {
+				contents = new Buffer(this.toString());
+			}
 			if (this.bom) {
-				return Buffer.concat([this.bom, contents]);
+				contents = Buffer.concat([this.bom, contents]);
 			}
 			return contents;
 		}

--- a/package.json
+++ b/package.json
@@ -40,7 +40,7 @@
     "through2": "^0.6.3"
   },
   "dependencies": {
-    "buffer-equals": "^1.0.3",
+    "buffer-equals": "^1.0.4",
     "iconv-lite": "^0.4.15"
   }
 }

--- a/package.json
+++ b/package.json
@@ -41,6 +41,6 @@
   },
   "dependencies": {
     "buffer-equals": "^1.0.3",
-    "iconv-lite": "0.4.11"
+    "iconv-lite": "^0.4.15"
   }
 }

